### PR TITLE
fix(facebook): postlink crash

### LINKED ISSUE #4793: [Crash] TypeError: Invalid Username
I was ,just browsing, then Mask reports an error.

> Invalid Username

Error stack:

<pre>TypeError: Invalid Username
    at getPostUrlAtFacebook (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8939.js:1239:49)
    at Object.getPostURL [as getURLFromPostIdentifier] (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8939.js:21:200)
    at Object.getCurrentValue (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8446.js:5980:119)
    at DecryptPost.ref (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8446.js:5425:44)
    at pj (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:248:366)
    at dk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:295:100)
    at ck (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:277:86)
    at J (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/2430.js:16900:203)
    at MessagePort.R (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/2430.js:16901:128)</pre>

## Build info
- Version: 2.0.2
- NODE_ENV: production
- target: chromium
- build: stable
- architecture: web
- BUILD_DATE: 2021-11-02T11:15:35.966Z
- VERSION: v1.29.12-1296-gf03237e1f

## Git ():

f03237e1f (released) on tag "v2.0.2"
git@github.com:DimensionDev/Maskbook.git

### LINKED ISSUE #4884: [Crash] TypeError: Invalid Username
I was scrolling in the one of facebook groups, then Mask reports an error.
error_boundary_crash_title above post.

> Invalid Username

Error stack:

<pre>TypeError: Invalid Username
    at getPostUrlAtFacebook (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8939.js:1239:49)
    at Object.getPostURL [as getURLFromPostIdentifier] (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8939.js:21:200)
    at Object.getCurrentValue (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8446.js:5980:119)
    at DecryptPost.ref (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/chunk.8446.js:5425:44)
    at pj (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:248:366)
    at dk (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:295:100)
    at ck (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/npm.react-dom.js:277:86)
    at J (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/2430.js:16900:203)
    at MessagePort.R (chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/js/2430.js:16901:128)</pre>

## Build info
- Version: 2.0.2
- NODE_ENV: production
- target: chromium
- build: stable
- architecture: web
- BUILD_DATE: 2021-11-02T11:15:35.966Z
- VERSION: v1.29.12-1296-gf03237e1f

## Git ():

f03237e1f (released) on tag "v2.0.2"
git@github.com:DimensionDev/Maskbook.git

### PR DESCRIPTION
**Pull request by zhouhanseng**
closes #4884
closes #4793

### CODE REVIEW

**Comment by zhouhanseng** [comment]
SNS post url only utilizes to append the post link to the sharing text. E.g. lucky drop claiming successful share.

```
I just claimed a lucky drop from @Zhouhancheng on Polygon network. Follow @realMaskNetwork (mask.io) to claim lucky drops.
#mask_io #RedPacket
https://twitter.com/Zhouhancheng/status/1463064144391770112 
```

If post link is broken:
```
I just claimed a lucky drop from @Zhouhancheng on Polygon network. Follow @realMaskNetwork (mask.io) to claim lucky drops.
#mask_io #RedPacket
```

Since it doesn't matter too much, we should not pre-throw a crash error.

File: packages/mask/src/social-network-adaptor/facebook.com/utils/parse-username.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export function isValidFacebookUsername(name: string) {
     // Avoid common mistake
     if (name === 'photo.php') return null
     const n = name.toLowerCase().replace(/\./g, '')
-    if (n.match(/^[\da-z]{5,}$/)) {
+    if (n.match(/^[\da-z]+$/)) {
```
Comment (zhouhanseng) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/parse-username.ts:14]: https://www.facebook.com/nba

`'nba'.match(/^[\da-z]+$/) // null`

**Review by guanbinrui — APPROVED** [review]

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Comment by wenluomask** [comment]
Error still exists, check https://www.facebook.com/permalink.php?story_fbid=2077533122422690&id=437617976414221.

**Comment by zhouhanseng** [comment]
@wenluomask fix

**Review by Jack-Works — APPROVED** [review]

**Comment by wenluomask** [comment]
Checked.

**Review by yanzhihong23 — APPROVED** [review]

**Comment by vietanh2131** [comment]
when will this fix be released?

**Comment by Jack-Works** [comment]
> when will this fix be released?

In our next version. @guanbinrui is it in #4780?

**Comment by guanbinrui** [comment]
> > when will this fix be released?
> 
> 
> 
> In our next version. @guanbinrui is it in #4780?

It's in our next version.

**Pull request merged**